rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchDomain.scala

```diff
@@ -0,0 +1,333 @@
+
+  //Parameters
+  final case object ParameterName  extends QSAttribute {
+    override val name = "param_name"
+    override val display = "Parameter name"
+  }
+  final case object ParameterValue extends QSAttribute {
+    override val name = "param_value"
+    override val display = "Parameter value"
+  }
+
+  //Rules
+  final case object RuleId       extends QSAttribute {
+    override val name = "id"
+    override val display = "ID"
+  }
+  final case object DirectiveIds extends QSAttribute { override val name = "directives" }
+  final case object Targets      extends QSAttribute { override val name = "groups" }
+
+  ////// all attributes /////
+  final val all: Set[QSAttribute] = sealerate.values[QSAttribute]
+
+}
+
+/*
+ * Objects on which we are able to perform
+ * quicksearch
+ */
+sealed trait QSObject { def name: String; def attributes: Set[QSAttribute] }
+
+object QSObject {
+  import QSAttribute._
+
+  final case object Common      extends QSObject { override val name = "common"
+                                                   override val attributes : Set[QSAttribute] = Set(Name, Description, LongDescription, IsEnabled)
+  }
+
+  final case object Node      extends QSObject { override val name = "node"
+                                                 override val attributes : Set[QSAttribute] = Set(NodeId, Fqdn, OsType, OsName
+                                                   , OsVersion, OsFullName, OsKernelVersion, OsServicePack, Arch, Ram
+                                                   , IpAddresses, PolicyServerId, Properties, RudderRoles)
+  }
+  final case object Group     extends QSObject { override val name = "group"
+                                                 override val attributes : Set[QSAttribute] = Set(GroupId, IsDynamic)
+  }
+  final case object Directive extends QSObject { override val name = "directive"
+                                                 override val attributes : Set[QSAttribute] = Set(DirectiveId, DirectiveVarName
+                                                   , DirectiveVarValue, TechniqueName, TechniqueId, TechniqueVersion)
+  }
+  final case object Parameter extends QSObject { override val name = "parameter"
+                                                 override val attributes : Set[QSAttribute] = Set(ParameterName, ParameterValue)
+  }
+  final case object Rule      extends QSObject { override val name = "rule"
+                                                 override val attributes : Set[QSAttribute] = Set(RuleId, DirectiveIds, Targets)
+  }
+
+  final val all: Set[QSObject] = sealerate.values[QSObject]
+
+  // default sort for QuickSearchResult:
+  // - by type
+  // - then by name
+  def sortQSObject(a: QSObject, b:QSObject): Boolean = {
+
+    implicit class QSObjectOrder(o: QSObject) {
+      def order() = o match {
+        case Common    => 0
+        case Node      => 1
+        case Group     => 2
+        case Directive => 3
+        case Parameter => 4
```

Comment:
Could you inverse param and directive here ? (ie directive => 4, param => 3). Because param are used in directive